src/java/org/apache/cassandra/locator/SimpleSeedProvider.java

```diff
@@ -62,6 +105,12 @@ public List<InetAddressAndPort> getSeeds()
                 logger.warn("Seed provider couldn't lookup host {}", host);
             }
         }
+
+        if (seeds.size() >= SEED_COUNT_WARN_THRESHOLD)
+            logger.warn("Seed provider returned more than {} seeds. " +
+                        "A large seed list may impact effectiveness of the third gossip round",
+                        SEED_COUNT_WARN_THRESHOLD);
```

Comment:
I understand it would be an overkill to use a guardrail for this, given that this isn't a limit for user actions but just a warning for admins.